packages/gatsby-transformer-json/src/gatsby-node.js

```diff
@@ -42,7 +42,12 @@ async function onCreateNode(
   }
 
   const content = await loadNodeContent(node)
-  const parsedContent = JSON.parse(content)
+  let parsedContent
+  try {
+    parsedContent = JSON.parse(content)
+  } catch {
+    throw new Error(`Unable to parse JSON: ${node.absolutePath}`)
```

Comment:
`node` is not guaranteed to be `File` node, so there might not be `absolutePath`. Take a look at similar messaging we have in https://github.com/gatsbyjs/gatsby/blob/88056f054cb6a539f5cb371336174516e47a1c07/packages/gatsby-transformer-remark/src/on-node-create.js#L69-L74

I think we could use something similar here

Reply:
Oh thanks for pointing that out, I updated the patch with this in mind.